onnx/defs/data_propagators.h

```diff
@@ -0,0 +1,32 @@
+#include "onnx/defs/shape_inference.h"
+
+namespace ONNX_NAMESPACE {
+
+// Please note that TensorShapeProto only supports integer shape
```

Comment:
Not sure whether this comment is needed or it should be improved

Reply:
Attaching a more complete comment below. But I think a better place for this would be where
DataPropagationContext is defined.
```
We use data propagation to perform partial evaluation of the model, to compute statically
known information about tensor values. It is intended to improve the precision of shape
inference. We reuse TensorShapeProto to represent the statically known values. One
limitation of this is that TensorShapeProto can represent only integer values.
As an example, data-propagation is intended to handle code-fragments like below:
   shape = Shape(X)
   batchsize = Slice(shape, [0], [1])
   newshape = Concat (batchsize, [1024, 1024])
   Z = Reshape(Y, newshape)
```

Reply:
Updated. @gramalingam Thank you for the detailed explanation!